jni/src/faiss_wrapper.cpp

```diff
@@ -28,9 +28,33 @@
 
 // Defines type of IDSelector
 enum FilterIdsSelectorType{
-    BITMAP, BATCH
+    BITMAP = 0, BATCH = 1,
 };
-
+namespace faiss {
+struct IDSelectorJlongBitmap : IDSelector {
+    size_t n;
+    const jlong* bitmap;
+
+    /** Construct with a binary mask like Lucene FixedBitSet
+     *
+     * @param n size of the bitmap array
+     * @param bitmap id like Lucene FixedBitSet bits
+     */
+    IDSelectorJlongBitmap(size_t n, const jlong* bitmap) : n(n), bitmap(bitmap) {};
+    bool is_member(idx_t id) const final {
+        uint64_t index = id;
+        uint64_t i = index >> 6;  // div 64
+        if (i >= n ) {
+            return false;
+        }
+        // signed shift will keep a negative index and force an
+        // array-index-out-of-bounds-exception, removing the need for an explicit check.
+        jlong bitmask = 1L << index;
+        return (bitmap[i] & bitmask) != 0;
+    }
+    ~IDSelectorJlongBitmap() override {}
```

Comment:
what will happen in the case if Lucene moves to other implementation of bitset where instead of using 64 bits they start to use some thing else , or change the implementation?

How the code is able to detect those changes?

Reply:
> what will happen in the case if Lucene moves to other implementation of bitset where instead of using 64 bits they start to use some thing else , or change the implementation?
> 
> How the code is able to detect those changes?

When LuceneFixedBitSet Changed, we must update lucene maven and do tests i have UnitTest for this: https://github.com/opensearch-project/k-NN/blob/bfbfb55ab8eaedee1e19894ab5001e3339e9c00a/src/test/java/org/opensearch/knn/jni/JNIServiceTests.java#L746-L758

when the test failed, we can change the implementation.